src/services/actions/CliActions.ts

```diff
@@ -224,6 +227,53 @@ export class CliActions {
     }
   }
 
+  /**
+  * Upgrades an app to a newer version available in the app catalog.
+  *
+  * @param node The tree item representing the app to be upgraded.
+  */
+  public static async upgradeAppCatalogApp(node: ActionTreeItem) {
+    try {
+      const actionNode = node.children?.find(child => child.contextValue === ContextKeys.removeApp);
```

Comment:
shouldn't this be `upgradeApp`

```suggestion
      const actionNode = node.children?.find(child => child.contextValue === ContextKeys.upgradeApp);
```